File: Mage/CoreData/ObservationImportant.m
//
//  ObservationImportant+CoreDataClass.m
//  mage-ios-sdk
//
//  Created by William Newman on 9/19/16.
//  Copyright © 2016 National Geospatial-Intelligence Agency. All rights reserved.
//

#import "ObservationImportant.h"
#import "NSDate+Iso8601.h"

@implementation ObservationImportant


+ (ObservationImportant *) importantForJson: (NSDictionary *) json inManagedObjectContext:(NSManagedObjectContext *) context {
    ObservationImportant *important = [ObservationImportant MR_createEntityInContext:context];
    [important updateImportantForJson:json];
    
    return important;
}

- (void) updateImportantForJson: (NSDictionary *) json {
    self.dirty = [NSNumber numberWithBool:NO];
    self.important = [NSNumber numberWithBool:YES];
    self.userId = [json objectForKey:@"userId"];
    self.timestamp = [NSDate dateFromIso8601String:[json objectForKey:@"timestamp"]];
    self.reason = [json objectForKey:@"description"];
}


@end

File: Mage/CoreData/ObservationImportant.swift
//
//  ObservationImportant+CoreDataClass.m
//  mage-ios-sdk
//
//  Created by William Newman on 9/19/16.
//  Copyright © 2016 National Geospatial-Intelligence Agency. All rights reserved.
//

import Foundation
import CoreData

@objc public class ObservationImportant : NSManagedObject {
    
    @objc public static func important(json: [String : Any], context: NSManagedObjectContext) -> ObservationImportant? {
        let important = ObservationImportant(context: context);
        important.update(json: json);
        try? context.obtainPermanentIDs(for: [important])
        return important
    }
    
    @objc public func update(json: [String : Any]) {
        self.dirty = false
        self.important = true
        self.userId = json[ObservationImportantKey.userId.key] as? String
        self.reason = json[ObservationImportantKey.description.key] as? String;
        
        if let timestamp = json[ObservationImportantKey.timestamp.key] as? String {
            self.timestamp = Date.ISO8601FormatStyle.gmtZeroDate(from: timestamp);
        }
    }
}
